From khali@linux-fr.org Mon Oct 17 14:10:53 2005
Date: Mon, 17 Oct 2005 23:07:05 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Deepak Saxena <dsaxena@plexity.net>
Subject: [PATCH 09/14] i2c: kzalloc conversion, ixp bus drivers
Message-Id: <20051017230705.1aac3bf2.khali@linux-fr.org>
Content-Disposition: inline; filename=kzalloc-01-i2c-ixp.patch

From: Deepak Saxena <dsaxena@plexity.net>

Use kzalloc instead of kmalloc+memzero in the ixp2000 and ixp4xx
I2C bus drivers.

Signed-off-by: Deepak Saxena <dsaxena@mvista.com>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/busses/i2c-ixp2000.c |    3 +--
 drivers/i2c/busses/i2c-ixp4xx.c  |    3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/i2c/busses/i2c-ixp2000.c
+++ gregkh-2.6/drivers/i2c/busses/i2c-ixp2000.c
@@ -106,11 +106,10 @@ static int ixp2000_i2c_probe(struct devi
 	struct platform_device *plat_dev = to_platform_device(dev);
 	struct ixp2000_i2c_pins *gpio = plat_dev->dev.platform_data;
 	struct ixp2000_i2c_data *drv_data = 
-		kmalloc(sizeof(struct ixp2000_i2c_data), GFP_KERNEL);
+		kzalloc(sizeof(struct ixp2000_i2c_data), GFP_KERNEL);
 
 	if (!drv_data)
 		return -ENOMEM;
-	memzero(drv_data, sizeof(*drv_data));
 	drv_data->gpio_pins = gpio;
 
 	drv_data->algo_data.data = gpio;
--- gregkh-2.6.orig/drivers/i2c/busses/i2c-ixp4xx.c
+++ gregkh-2.6/drivers/i2c/busses/i2c-ixp4xx.c
@@ -107,12 +107,11 @@ static int ixp4xx_i2c_probe(struct devic
 	struct platform_device *plat_dev = to_platform_device(dev);
 	struct ixp4xx_i2c_pins *gpio = plat_dev->dev.platform_data;
 	struct ixp4xx_i2c_data *drv_data = 
-		kmalloc(sizeof(struct ixp4xx_i2c_data), GFP_KERNEL);
+		kzalloc(sizeof(struct ixp4xx_i2c_data), GFP_KERNEL);
 
 	if(!drv_data)
 		return -ENOMEM;
 
-	memzero(drv_data, sizeof(struct ixp4xx_i2c_data));
 	drv_data->gpio_pins = gpio;
 
 	/*
